gnovm/cmd/gno/repl.go

```diff
@@ -91,68 +113,196 @@ func runRepl(cfg *replCfg) error {
 		handleInput(r, cfg.initialCommand)
 	}
 
-	fmt.Fprint(os.Stdout, "gno> ")
+	fmt.Fprint(os.Stdout, gnoREPL)
+
+	var (
+		inEdit      bool
+		prev        string
+		indentLevel int
+	)
 
-	inEdit := false
-	prev := ""
 	liner := bufio.NewScanner(os.Stdin)
 
 	for liner.Scan() {
 		line := liner.Text()
 
-		if l := strings.TrimSpace(line); l == ";" {
-			line, inEdit = "", false
-		} else if l == "/editor" {
-			line, inEdit = "", true
-			fmt.Fprintln(os.Stdout, "// enter a single ';' to quit and commit")
-		}
+		trimmedLine := strings.TrimSpace(line)
+
+		indentLevel = updateIndentLevel(trimmedLine, indentLevel)
+		line, inEdit = handleEditor(line)
```

Comment:
This change breaks the editor mode. I'd recommend restoring it since it is only called from this one location and moving it to its own function doesn't have too much benefit. The issue can by entering editor mode with `/editor` and then doing some variable assignments. Notice that editor mode is exited after doing a single variable assignment when it should remain in editor mode until the semicolon is encountered.